source/_posts/2018/2018-03-08-why-is-collector-pattern-so-awesome.md

```diff
@@ -0,0 +1,382 @@
+---
+id: 80
+title: "Why is Collector Pattern so Awesome"
+perex: '''
+    How to achieve *open for extension* and *closed for modification* [one of sOlid principals](https://github.com/jupeter/clean-code-php#openclosed-principle-ocp)?
```

Comment:
```diff
-one of sOlid principals
+one of solid principals
```

Reply:
That's intentional to refer to O, aka open closed